### PR TITLE
DOC: better intro for dates.py

### PR DESCRIPTION
**Pull request by jklymak**
## PR Summary

This attempts to improve the dates.py intro to give the user some idea what is going on.  This page comes up first if you type `matplotlib dates` into Google, but I don't think anyone would know how to use dates from the original. 

## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

**Comment by jklymak** [comment]
Result: https://54191-1385122-gh.circle-artifacts.com/0/doc/build/html/api/dates_api.html

File: lib/matplotlib/dates.py [hunk 1]
```diff
@@ -2,6 +2,26 @@
 Matplotlib provides sophisticated date plotting capabilities, standing on the
 shoulders of python :mod:`datetime` and the add-on module :mod:`dateutil`.
 
+By default, Matpltolib uses the units machinery described in
```
Comment (QuLogic) [inline lib/matplotlib/dates.py:5]:
```suggestion
By default, Matplotlib uses the units machinery described in
```

File: lib/matplotlib/dates.py [hunk 2]
```diff
@@ -2,6 +2,26 @@
...
+`~.matplotlib.units` to convert `datetime.datetime`, and `numpy.datetime64`
```
Comment (QuLogic) [inline lib/matplotlib/dates.py:6]: `matplotlib` is top-level; don't need a `.` prefix.
```suggestion
`~matplotlib.units` to convert `datetime.datetime`, and `numpy.datetime64`
```

File: lib/matplotlib/dates.py [hunk 3]
```diff
@@ -2,6 +2,26 @@
...
+need to do anything for dates to be formatted, but dates often have strict
+formatting needs, so this module provides many axis locators and formatters.
+A basic example using `numpy.datetime64` is::
+
+    import numpy as np
+    fig, ax = plt.subplots()
+    times = np.arange(np.datetime64('2001-01-02'),
+                      np.datetime64('2002-02-03'), np.timedelta64(75, 'm'))
+    y = np.random.randn(len(times))
+    ax.plot(times, y)
```
Comment (QuLogic) [inline lib/matplotlib/dates.py:17]: I would not mix data generation and plotting:
```suggestion

    times = np.arange(np.datetime64('2001-01-02'),
                      np.datetime64('2002-02-03'), np.timedelta64(75, 'm'))
    y = np.random.randn(len(times))

    fig, ax = plt.subplots()
    ax.plot(times, y)
```

**Review by QuLogic — APPROVED** [review]

**Pull request merged**